File: desktop/test/censorshipVerdict.test.ts
import { describe, expect, it } from 'vitest';
import { evaluateCensorship } from '../src/shared/censorshipVerdict';

describe('evaluateCensorship', () => {
  it('both reachable means OK', () => {
    expect(evaluateCensorship(true, true)).toBe('OK');
  });

  it('whitelist reachable but foreign blocked means operator restriction', () => {
    expect(evaluateCensorship(true, false)).toBe('OPERATOR_RESTRICTION');
  });

  it('neither reachable means generic connectivity problem', () => {
    expect(evaluateCensorship(false, false)).toBe('NO_CONNECTIVITY');
  });

  it('foreign reachable alone is still OK', () => {
    expect(evaluateCensorship(false, true)).toBe('OK');
  });
});

File: android/app/src/test/java/com/vpn/android/vpn/CensorshipVerdictTest.java
package com.vpn.android.vpn;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

/**
 * Covers the "honest blocking screen" decision from docs/PLAN.md §6: a
 * reachable whitelisted host plus an unreachable foreign host must be read
 * as an operator restriction, not a generic app error.
 */
public class CensorshipVerdictTest {

    @Test
    public void bothReachableMeansOk() {
        assertEquals(CensorshipVerdict.Result.OK, CensorshipVerdict.evaluate(true, true));
    }

    @Test
    public void whitelistReachableButForeignBlockedMeansOperatorRestriction() {
        assertEquals(CensorshipVerdict.Result.OPERATOR_RESTRICTION, CensorshipVerdict.evaluate(true, false));
    }

    @Test
    public void neitherReachableMeansGenericConnectivityProblem() {
        assertEquals(CensorshipVerdict.Result.NO_CONNECTIVITY, CensorshipVerdict.evaluate(false, false));
    }

    @Test
    public void foreignReachableAloneIsStillOk() {
        // The target we actually care about works; a flaky/unreachable whitelist probe
        // host must not itself produce a false "no connectivity" reading.
        assertEquals(CensorshipVerdict.Result.OK, CensorshipVerdict.evaluate(false, true));
    }
}
